dash/include/dash/halo/HaloStencilOperator.h

```diff
@@ -0,0 +1,267 @@
+          _halo_block.boundary_size()) {}
+
+  HaloStencilOperator() = delete;
+
+  /// returns the begin iterator for all relevant elements (inner + boundary)
+  iterator begin() noexcept { return _begin; }
+
+  /// returns the begin const iterator for all relevant elements
+  /// (inner + boundary)
+  const_iterator begin() const noexcept { return _begin; }
+
+  /// returns the end iterator for all relevant elements (inner + boundary)
+  iterator end() noexcept { return _end; }
+
+  /// returns the end const iterator for all relevant elements (inner +
+  /// boundary)
+  const_iterator end() const noexcept { return _end; }
+
+  /// returns the begin iterator for all inner elements
+  iterator_inner ibegin() noexcept { return _ibegin; }
+
+  /// returns the begin const iterator for all inner elements
+  const_iterator_inner ibegin() const noexcept { return _ibegin; }
+
+  /// returns the end iterator for all inner elements
+  iterator_inner iend() noexcept { return _iend; }
+
+  /// returns the end const iterator for all inner elements
+  const_iterator_inner iend() const noexcept { return _iend; }
+
+  /// returns the begin iterator for all boundary elements
+  iterator_bnd bbegin() noexcept { return _bbegin; }
+
+  /// returns the begin const iterator for all boundary elements
+  const_iterator_bnd bbegin() const noexcept { return _bbegin; }
+
+  /// returns the end iterator for all boundary elements
+  iterator_bnd bend() noexcept { return _bend; }
+
+  /// returns the end const iterator for all boundary elements
+  const_iterator_bnd bend() const noexcept { return _bend; }
+
+  /**
+   * Returns the \ref HaloBlock
+   */
+  const HaloBlock_t& halo_block() { return _halo_block; }
+
+  /**
+   * Returns the stencil specification \ref StencilSpec
+   */
+  const StencilSpecT& stencil_spec() const { return _stencil_spec; }
+
+  /**
+   * Returns the halo memory management object \ref HaloMemory
+   */
+  HaloMemory_t& halo_memory() { return _halo_memory; }
+
+  /**
+   * Modifies all stencil point elements and the center within the inner view.
+   * The stencil points are multiplied with their coefficent (\ref StencilPoint)
+   * and the center is  multiplied with the given center coefficient. The
+   * results then modifies the center/stencil point elements via the given
+   * operation.
+   *
+   * \param coords center coordinate
+   * \param value base value for all points
+   * \param coefficient for center
+   * \param op operation to use (e.g. std::plus). default: replace
+   */
+  void set_value_at_inner_local(
+    const ElementCoords_t& coords, ElementT value, ElementT coefficient_center,
+    std::function<ElementT(const ElementT&, const ElementT&)> op =
+      [](const ElementT& lhs, const ElementT& rhs) { return rhs; }) {
+    auto*           center = _local_memory;
+    pattern_index_t offset = 0;
+
+    if(MemoryArrange == ROW_MAJOR) {
+      offset = coords[0];
+      for(auto d = 1; d < NumDimensions; ++d)
```

Comment:
This `auto` does maynot result in the same type as `NumStencilPoints`. This issue comes up multiple times in this file.